astropy/stats/nanfunctions.py

```diff
@@ -77,7 +77,11 @@ def _apply_bottleneck(
 
         result = function(array, axis=axis, **kwargs)
         if isinstance(array, Quantity):
-            return array.__array_wrap__(result)
+            if function == bottleneck.nanvar:
+                result <<= array.unit**2
```

Comment:
We know here that `array` is a `Quantity`, so can use the private method `array._result_as_quantity(result, unit**2)` - that would keep any subclass in place but ensure that, e.g., an `Angle` will drop down to a `Quantity`.

Reply:
Thanks, @mhvk! I've updated this to use `_result_as_quantity` and added a test for `Angle` input.